internal/http/services/owncloud/ocdav/tus.go

```diff
@@ -158,6 +168,43 @@ func (s *svc) handleTusPost(ctx context.Context, w http.ResponseWriter, r *http.
 				return
 			}
 		}
+		if isSecretFileDrop {
+			lReq := &provider.ListContainerRequest{
+				Ref: &provider.Reference{
+					ResourceId: sRes.GetInfo().GetParentId(),
+				},
+			}
+			lRes, err := client.ListContainer(ctx, lReq)
+			if err != nil {
+				log.Error().Err(err).Msg("error sending grpc stat request")
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			if lRes.Status.Code != rpc.Code_CODE_OK {
+				log.Debug().Err(err).Msg("error listing container")
+				errors.HandleErrorStatus(&log, w, lRes.Status)
```

Comment:
Better to avoid returning the status code directly. Could be `PermissionDenied` vs `NotFound`. Better to return `InternalServerError` always or explicitly check for specific status codes

Reply:
HandleErrorStatus is used to handle the different possible status codes. It correctly converts the mto HTTP status codes.